Marlin/src/HAL/AVR/fast_pwm.cpp

```diff
@@ -162,30 +166,35 @@ void set_pwm_frequency(const pin_t pin, const uint16_t f_desired) {
     }
   }
 
-  _SET_WGMnQ(timer.TCCRnQ, wgm);
-  _SET_CSn(timer.TCCRnQ, j);
+  _SET_WGMnQ(timer, wgm);
+  _SET_CSn(timer, j);
 
   if (is_timer2) {
-    TERN_(USE_OCR2A_AS_TOP, _SET_OCRnQ(timer.OCRnQ, 0, res)); // Set OCR2A value (TOP) = res
+    TERN_(USE_OCR2A_AS_TOP, _SET_OCRnQ(timer, 0, res)); // Set OCR2A value (TOP) = res
   }
   else
-    _SET_ICRn(timer.ICRn, res);                               // Set ICRn value (TOP) = res
+    _SET_ICRn(timer, res);                              // Set ICRn value (TOP) = res
 }
 
 void set_pwm_duty(const pin_t pin, const uint16_t v, const uint16_t v_size/*=255*/, const bool invert/*=false*/) {
   // If v is 0 or v_size (max), digitalWrite to LOW or HIGH.
-  // Note that digitalWrite also disables pwm output for us (sets COM bit to 0)
+  // Note that digitalWrite also disables PWM output for us (sets COM bit to 0)
   if (v == 0)
     digitalWrite(pin, invert);
   else if (v == v_size)
     digitalWrite(pin, !invert);
   else {
-    Timer timer = get_pwm_timer(pin);
-    if (timer.isProtected) return;                            // Leave protected timer unchanged
+    const Timer timer = get_pwm_timer(pin);
     if (timer.isPWM) {
-      _SET_COMnQ(timer.TCCRnQ, SUM_TERN(HAS_TCCR2, timer.q, timer.q == 2), COM_CLEAR_SET + invert);   // COM20 is on bit 4 of TCCR2, so +1 for q==2
-      const uint16_t top = timer.n == 2 ? TERN(USE_OCR2A_AS_TOP, *timer.OCRnQ[0], 255) : *timer.ICRn;
-      _SET_OCRnQ(timer.OCRnQ, timer.q, uint16_t(uint32_t(v) * top / v_size)); // Scale 8/16-bit v to top value
+      if (timer.n == 0) {
+        _SET_COMnQ(timer, timer.q, COM_CLEAR_SET);  // Only allow a TIMER0B select...
```

Comment:
Definitely no application of `invert` here?

Reply:
I also note that this timer is marked as protected, so `set_pwm_frequency` never touches it, but `set_pwm_duty` is allowed to modify it. Is it not safe to initialize it with `set_pwm_frequency` ?

Reply:
Oh I see you mean timer0 has no invert, correct.